### PR TITLE
LPS-174104 Null when accessing Object Entry with a One to Many relationship with System Object after Upgrade

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-174104](https://issues.liferay.com/browse/LPS-174104)

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v4_2_0/ObjectFieldUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * @author Pedro Leite
+ */
+public class ObjectFieldUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		String selectSQL = SQLTransformer.transform(
+			StringBundler.concat(
```
Comment (p-albuquerque) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v4_2_0/ObjectFieldUpgradeProcess.java:35]: Use the order of columns within the entity definition in `service.xml` to sort the columns in select statement
```
    select ObjectField.objectFieldId, ObjectField.companyId,
    ObjectField.dbTableName ...
```

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:objects

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v4_2_0/ObjectFieldUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				selectSQL);
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update ObjectField set dbTableName = ? where " +
+						"objectFieldId = ?");
+			ResultSet resultSet = preparedStatement1.executeQuery()) {
+
+			while (resultSet.next()) {
+				String sufixDbTableName =
```
Comment (p-albuquerque) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v4_2_0/ObjectFieldUpgradeProcess.java:56]: Rename **sufixDbTableName** to **suffixDBTableName**

Comment (p-albuquerque) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v4_2_0/ObjectFieldUpgradeProcess.java:56]: Rename **sufixDbTableName** to **suffixDBTableName**

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by p-albuquerque — APPROVED** [review]

**Comment by liferay-peds** [comment]
CC: @pedro-oliveira446

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:forward

**Pull request closed without merging**